contract_sale_invoicing/models/contract.py

```diff
@@ -16,16 +16,84 @@ class ContractContract(models.Model):
     @api.multi
     def _recurring_create_invoice(self, date_ref=False):
         invoices = super()._recurring_create_invoice(date_ref)
-        if not self.invoicing_sales:
-            return invoices
-        sales = self.env['sale.order'].search([
-            ('analytic_account_id', '=', self.group_id.id),
-            ('partner_invoice_id', 'child_of',
-             self.partner_id.commercial_partner_id.ids),
-            ('invoice_status', '=', 'to invoice'),
-            ('date_order', '<=',
-             '{} 23:59:59'.format(self.recurring_next_date)),
-        ])
-        if sales:
-            invoice_ids = sales.action_invoice_create()
-            invoices |= self.env['account.invoice'].browse(invoice_ids)[:1]
+        for rec in self:
+            if not rec.invoicing_sales:
+                return invoices
+            sales = self.env['sale.order'].search([
+                ('analytic_account_id', '=', rec.group_id.id),
```

Comment:
On the contract, when the checkbox is True, we should display a selection field "Group By" with 2 options: "Analytic Account", "Contract" and use the value to search for analytic account or contract.

Reply:
I would get the domain from a new method so it is easier to extend.

Reply:
@max3903 @sbejaoui 
I Add one selection fields `Group By` with 2 options: "Analytic Account", "Contract" and when a select checkbox then visible selection field. 
Right?

Reply:
@nikul-serpentcs Yes, see https://github.com/nikul-serpentcs/contract/pull/2

Reply:
> @nikul-serpentcs @murtuzasaleh On the contract, when the checkbox is True, we need to add a selection field "Group By" with 2 options: "Analytic Account", "Contract" and use the value here to search for analytic accounts or contracts. Default value should be "Analytic Account".
> 
> CC: @sbejaoui

why not getting both. for a given contract, get all linked SO whatever the link, contract and analytic account

Reply:
@sbejaoui In our situation, all the contracts are linked to the same analytic account.